### ISSUE TITLE
Problem with git.latest state after upgrade to 2015.8.8.2 

### ISSUE THREAD

**Issue by zerthimon** [saltstack/salt#32237]
After upgrading to 2015.8.8.2 from 2015.8.7 I'm having the following error:
```
[INFO    ] Executing command ['git', 'ls-remote', 'git@my_project:my_project/configs.git'] as user 'root' in directory '/root'
[ERROR   ] Command ['git', 'ls-remote', 'git@my_project:my_project/configs.git'] failed with return code: 128
[ERROR   ] stderr: ssh: Could not resolve hostname my_project: Name or service not known
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
[ERROR   ] retcode: 128
[ERROR   ] Failed to check remote refs: Unable to authenticate using identity file:

ssh: Could not resolve hostname my_project: Name or service not known
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
[INFO    ] Completed state [git@my_project:my_project/configs.git] at time 15:36:54.773503
```
Running the same command in shell works fine:
```
# git ls-remote git@my_project:my_project/configs.git
a43a1dabf0e4c52de98497ad9f8ce78196d6da66	HEAD
461d4a940ba53af52ba0b4b5d874c5f388a94d97	refs/heads/CORE-150-build-static
69a2a7bfb02f34c7eba0f7eb10551d8c3c0da969	refs/heads/CORE-37
```
State:
```
my_project_repo:
  git.latest:
    - name: git@my_project:my_project/configs.git
    - target: /opt/projects/configs
    - rev: master
    - identity: /root/.ssh/configs.pkey
    - user: root
    - require:
      - pkg: git_package
      - ssh_known_hosts: my_project_known_hosts
      - file: my_project_directory
      - file: my_project_keyfile
      - file: my_project_ssh_config
```
versions:
```
# salt-call --versions-report
Salt Version:
           Salt: 2015.8.8.2
 
Dependency Versions:
         Jinja2: 2.7.2
       M2Crypto: Not Installed
           Mako: 0.9.1
         PyYAML: 3.10
          PyZMQ: 14.0.1
         Python: 2.7.6 (default, Jun 22 2015, 17:58:13)
           RAET: Not Installed
        Tornado: 4.3
            ZMQ: 4.0.4
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 1.5
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
        libgit2: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.3.0
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: Not Installed
        timelib: Not Installed
 
System Versions:
           dist: Ubuntu 14.04 trusty
        machine: x86_64
        release: 3.19.0-56-generic
         system: Ubuntu 14.04 trusty

```
Reverting back to 2015.8.7 fixes the issue.

**Comment by jfindlay** [comment]
@zerthimon, thanks for reporting.

**Comment by Kurocon** [comment]
Hello,

This also affects me on Debian after updating salt. Works in the terminal and after reverting salt to 2015.8.7 but is broken on 2015.8.8.2. The project the second error mentions exists on GitLab and is cloneable  manually with the identity file. The identity file that the fourth error mentions also exists and works when using it to manually clone the repository.

Error:
```
[INFO    ] Executing command ['git', 'ls-remote', 'ssh://git@git.someserver.nl:www/mysite.git'] as user 'www-data' in directory '/var/www'
[ERROR   ] Command ['git', 'ls-remote', 'ssh://git@git.someserver.nl:www/mysite.git'] failed with return code: 128
[ERROR   ] stderr: GitLab: The project you were looking for could not be found.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
[ERROR   ] retcode: 128
[ERROR   ] Failed to check remote refs: Unable to authenticate using identity file:

GitLab: The project you were looking for could not be found.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
[INFO    ] Completed state [ssh://git@git.someserver.nl:www/mysite.git] at time 23:00:35.496979
```

State:
```
mysite_git:
  git.latest:
    - name: ssh://git@git.someserver.nl:www/mysite.git
    - target: /var/www/mysite
    - branch: master
    - user: www-data
    - identity: /var/www/.ssh/minion_deploy
    - require:
      - sls: apache.common
      - pkg: apache2
      - pkg: git
      - file: /var/www/mysite
      - file: /var/www/.ssh/minion_deploy.pub
      - file: /var/www/.ssh/minion_deploy
    - watch_in:
      - service: apache2
```

Versions:
```
Salt Version:
           Salt: 2015.8.8.2
 
Dependency Versions:
         Jinja2: 2.7.3
       M2Crypto: 0.21.1
           Mako: 1.0.0
         PyYAML: 3.11
          PyZMQ: 14.4.0
         Python: 2.7.9 (default, Mar  1 2015, 12:57:24)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
           cffi: 0.8.6
       cherrypy: Not Installed
       dateutil: 2.2
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
        libgit2: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.2
   mysql-python: 1.2.3
      pycparser: 2.10
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: Not Installed
        timelib: Not Installed
 
System Versions:
           dist: debian 8.4 
        machine: x86_64
        release: 3.16.0-4-amd64
         system: debian 8.4
```

**Comment by terminalmage** [comment]
This seems to be a duplicate of #32685 which I fixed yesterday with #33152, still awaiting merge.

Is ``my_project`` here matched by a ``Host`` entry in ``/root/.ssh/config``, and not an actual DNS name?

**Comment by zerthimon** [comment]
Yep, my_project is a Host entry in /root/.ssh/config

**Comment by terminalmage** [comment]
@zerthimon then it is definitely the same issue I fixed, and the pull request I referenced earlier fixes this.

**Comment by rallytime** [comment]
Closing as this is a duplicate that has already been fixed. This fix is available in 2015.8.10 and newer.

If someone bumps into this again, let us know and we can take another look. Thanks!
